pkg/fleet/installer/msi/msiexec.go

```diff
@@ -367,6 +369,30 @@ func isRetryableExitCode(err error) bool {
 	return false
 }
 
+// isSuccessExitCode returns true if the exit code indicates the msiexec operation was successful
+//
+// https://learn.microsoft.com/en-us/windows/win32/msi/error-codes
+func isSuccessExitCode(err error) bool {
+	if err == nil {
+		// no error means success
+		return true
+	}
+
+	var exitError exitCodeError
+	if errors.As(err, &exitError) {
+		if exitError.ExitCode() == int(windows.ERROR_SUCCESS_REBOOT_REQUIRED) {
+			// 3010 - success but requires reboot
+			return true
+		} else if exitError.ExitCode() == int(windows.ERROR_SUCCESS_REBOOT_INITIATED) {
+			// 1641 - success but Windows will reboot the host
+			// this is unexpected now that we pass /norestart, msiexec should return 3010 instead
```

Comment:
Shouldn't we error on it then ..?

Reply:
it's better to treat it as success here, as the Agent has either been successfully uninstalled or successfully installed, and callers expect that returning an error means the opposite of that.

If we want to try to handle the sudden reboot better, then we can return an error in this case to signal to the callers. But since using `/norestart` means we're not expecting a sudden reboot anymore it shouldn't be an issue.

We'll still have the error reported by the `msiexec` span, as noted below.